addon-test-support/ember-mocha/setup-container-test.js

```diff
@@ -0,0 +1,43 @@
+import {
+  beforeEach,
+  afterEach
+} from 'mocha';
+import {
+  setupContext,
+  teardownContext
+} from '@ember/test-helpers';
+import { assign, merge } from '@ember/polyfills';
+
+const _assign = assign || merge;
+
+export default function setupUnitTest(options) {
+  let originalContext = {};
+
+  beforeEach(function() {
+    _assign(originalContext, this);
+
+    return setupContext(this, options).then(() => {
+
+      let originalPauseTest = this.pauseTest;
+      this.pauseTest = function Mocha_pauseTest() {
+        this.timeout(0); // prevent the test from timing out
+
+        return originalPauseTest.call(this);
+      };
+    });
+  });
+
+  afterEach(function() {
+    return teardownContext(this).then(() => {
+      // delete any extraneous properties
+      for (let key in this) {
+        if (!(key in originalContext)) {
+          delete this[key];
+        }
+      }
+
+      // copy over the original values
+      _assign(this, originalContext);
```

Comment:
do we have to reset `originalContext` to `{}` here?

Reply:
I just fixed this in `beforeEach`, to prevent accidental leaks. Though I don't think it makes much of a difference in practice. I tried for quite some time to find a test case that would have been affected by this, but somehow didn't succeed ;)